crypto-jni/src/lib.rs

```diff
@@ -408,3 +467,199 @@ pub extern "system" fn Java_com_concordium_sdk_crypto_CryptoJniNative_generateCo
 
     return CryptoJniResult::Ok(payload).to_jstring(&env);
 }
+
+type SerializeParamResult = CryptoJniResult<String>;
+
+#[no_mangle]
+#[allow(non_snake_case)]
+/**
+ * The JNI wrapper for serializing receive parameters.
+ * Constructs a SerializeParamResult containg the hex encoded serialized parameter. 
+ */
+pub extern "system" fn Java_com_concordium_sdk_crypto_CryptoJniNative_serializeReceiveParameter(
+    env: JNIEnv,
+    _: JClass,
+    parameter: JString,
+    contractName: JString,
+    methodName: JString,
+    schemaBytes: jbyteArray,
+    schemaVersion: jint,
+    verboseErrors: jboolean,
+) -> jstring {
+
+    let schema = match env.convert_byte_array(schemaBytes) {
+        Ok(x) => x,
+        Err(err) => return SerializeParamResult::from(err).to_jstring(&env),
+    };
+
+
+    let parameter: String = match env.get_string(parameter) {
+        Ok(java_str) => match java_str.to_str() {
+            Ok(rust_str) => String::from(rust_str),
+            Err(err) => return SerializeParamResult::from(err).to_jstring(&env),
+        },
+        Err(err) => return SerializeParamResult::from(err).to_jstring(&env),
+    };
+
+    let contractName: String = match env.get_string(contractName) {
+        Ok(java_str) => match java_str.to_str() {
+            Ok(rust_str) => String::from(rust_str),
+            Err(err) => return SerializeParamResult::from(err).to_jstring(&env),
+        },
+        Err(err) => return SerializeParamResult::from(err).to_jstring(&env),
+    };
+
+    let methodName: String = match env.get_string(methodName) {
+        Ok(java_str) => match java_str.to_str() {
+            Ok(rust_str) => String::from(rust_str),
+            Err(err) => return SerializeParamResult::from(err).to_jstring(&env),
+        },
+        Err(err) => return SerializeParamResult::from(err).to_jstring(&env),
+    };
+
+    let version: Option<u8> = match schemaVersion {
+        0 => Some(0),
+        1 => Some(1),
+        2 => Some(2),
+        3 => Some(3),
+        _ => None,
+    };
+
+    let verboseErrors = verboseErrors != JNI_FALSE;
+
+    let serializedParameter = serialize_receive_contract_parameters_aux(
+        &parameter, 
+        &contractName, 
+        &methodName, 
+        &schema, 
+        &version,
+        verboseErrors
+    );
+
+
+    let result = match serializedParameter {
+        Ok(serializedParameter) => hex::encode(serializedParameter),
+        Err(err) => return SerializeParamResult::from(err).to_jstring(&env),
+        
+    };
+
+    return SerializeParamResult::Ok(result).to_jstring(&env);
```

Comment:
So errors are also "Ok" ?

Reply:
Errors are returned as soon as they arise e.g. 
<img width="680" alt="image" src="https://github.com/Concordium/concordium-java-sdk/assets/93247743/90a67ddd-ff96-4163-a61e-ea8fcaf30eb1">
<img width="555" alt="image" src="https://github.com/Concordium/concordium-java-sdk/assets/93247743/e953929a-077b-4fb3-a702-a05d360e70ff">

If the line in question is ever excecuted the serialization was performed correctly